cosmrs/src/base/account_id.rs

```diff
@@ -79,7 +79,11 @@ impl FromStr for AccountId {
     type Err = ErrorReport;
 
     fn from_str(s: &str) -> Result<Self> {
-        let (hrp, bytes) = bech32::decode(s).wrap_err(format!("invalid bech32: '{}'", s))?;
+        let (hrp, bytes) = if s.starts_with(|c: char| c.is_uppercase()) {
+            bech32::decode_upper(s)
+        } else {
+            bech32::decode(s)
+        }.wrap_err(format!("invalid uppercase bech32: '{}'", s))?;
```

Comment:
@tony-iqlusion you need to fix the error message since it's now one only from `invalid uppercase` to `invalid`